stm32mp1.mk

```diff
@@ -0,0 +1,212 @@
+BINARIES_PATH		?= $(ROOT)/out/bin
+TF_A_PATH		?= $(ROOT)/trusted-firmware-a
+U_BOOT_PATH		?= $(ROOT)/u-boot
+
+define install_in_binaries
+	echo "  INSTALL $(shell basename $1) to $(BINARIES_PATH)" && \
+	mkdir -p $(BINARIES_PATH) && \
+	ln -sf $1 $(BINARIES_PATH)
+endef
+
+################################################################################
+# Main targets
+################################################################################
+all: arm-tf optee-os u-boot linux buildroot
+	@$(call install_in_binaries,$(ROOT)/out-br/images/sdcard.img)
+	@echo Build for platform $(PLATFORM) completed
+
+clean: arm-tf-clean optee-os-clean u-boot-clean linux-clean buildroot-clean \
+       check-clean
+
+include toolchain.mk
+
+################################################################################
+# OP-TEE OS
+################################################################################
+optee-os: optee-os-common
+	@$(call install_in_binaries,$(OPTEE_OS_PATH)/out/arm/core/$(OPTEE_HEADER_BIN))
+	@$(call install_in_binaries,$(OPTEE_OS_PATH)/out/arm/core/$(OPTEE_PAGER_BIN))
+	@$(call install_in_binaries,$(OPTEE_OS_PATH)/out/arm/core/$(OPTEE_PAGEABLE_BIN))
+
+optee-os-clean: optee-os-clean-common
+
+################################################################################
+# ARM Trusted Firmware
+################################################################################
+TF_A_EXPORTS ?= CROSS_COMPILE="$(CCACHE)$(AARCH32_CROSS_COMPILE)"
+
+TF_A_DEBUG ?= $(DEBUG)
+ifeq ($(TF_A_DEBUG),0)
+TF_A_LOGLVL ?= 30
+TF_A_OUT = $(TF_A_PATH)/build/stm32mp1/release
+else
+TF_A_LOGLVL ?= 50
+TF_A_OUT = $(TF_A_PATH)/build/stm32mp1/debug
+endif
+
+TF_A_FLAGS ?= \
+	BL32=$(BINARIES_PATH)/$(OPTEE_HEADER_BIN) \
+	BL32_EXTRA1=$(BINARIES_PATH)/$(OPTEE_PAGER_BIN) \
+	BL32_EXTRA2=$(BINARIES_PATH)/$(OPTEE_PAGEABLE_BIN) \
+	BL33=$(BINARIES_PATH)/$(U_BOOT_BIN) \
+	ARM_ARCH_MAJOR=7 \
+	ARCH=aarch32 \
+	PLAT=stm32mp1 \
+	DTB_FILE_NAME=$(STM32MP1_DTS_BASENAME).dtb \
+	AARCH32_SP=optee \
+	DEBUG=$(TF_A_DEBUG) \
+	LOG_LEVEL=$(TF_A_LOGLVL) \
+	STM32MP_EMMC=1 STM32MP_SDMMC=1 \
+	STM32MP_RAW_NAND=0 STM32MP_SPI_NAND=0 STM32MP_SPI_NOR=0
+
+arm-tf: optee-os u-boot
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) all
+	@$(call install_in_binaries,$(TF_A_OUT)/$(ARM_TF_BIN))
+
+arm-tf-clean:
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) clean
+
+################################################################################
+# U-Boot
+################################################################################
+U_BOOT_EXPORTS ?= CROSS_COMPILE="$(CCACHE)$(AARCH32_CROSS_COMPILE)"
+
+# Use stm32mp15_optee_defconfig up to U-Boot v2020.07-rc2.
+# Use stm32mp15_trusted_defconfig from v2020.07-rc3 onward.
+u-boot:
+ifneq ($(wildcard $(U_BOOT_PATH)/configs/stm32mp15_optee_defconfig),)
+	$(U_BOOT_EXPORTS) $(MAKE) -C $(U_BOOT_PATH) stm32mp15_optee_defconfig
+else
+	$(U_BOOT_EXPORTS) $(MAKE) -C $(U_BOOT_PATH) stm32mp15_trusted_defconfig
```

Comment:
Is there a guarantee that it'll find this one if the first "if" is false? Would it make sense to have another "if (..wildcard...)" and in the final "else" print an error message if no expected defconfig is found?

Reply:
If **stm32mp15_optee_defconfig** exists, it shall be used. If not, then, it's that this defconfig file was remove since merged into **stm32mp15_trusted_defconfig**.  So I expect the later is always found, but shall be used only if the former does not exist. U-boot config command should nicely fail if expected defconfig does not exist, which is not expected to occur.